examples/Console/TestInMemoryExporter.cs

```diff
@@ -0,0 +1,134 @@
+// <copyright file="TestInMemoryExporter.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using OpenTelemetry;
+using OpenTelemetry.Resources;
+using OpenTelemetry.Trace;
+
+namespace Examples.Console
+{
+    internal class TestInMemoryExporter
+    {
+        // To run this example, run the following command from
+        // the reporoot\examples\Console\.
+        // (eg: C:\repos\opentelemetry-dotnet\examples\Console\)
+        //
+        // dotnet run inmemory
+        internal static object Run(InMemoryOptions options)
+        {
+            // List that will be populated with the traces by InMemoryExporter
+            var exportedItems = new List<Activity>();
+
+            // Enable TracerProvider for the source "MyCompany.MyProduct.MyWebServer"
+            // and use a custom MyProcessor, along with InMemory Exporter.
+            using var tracerProvider = Sdk.CreateTracerProviderBuilder()
+                .AddSource("MyCompany.MyProduct.MyWebServer")
+                .SetResourceBuilder(ResourceBuilder.CreateDefault().AddService("MyServiceName"))
+                .AddProcessor(new MyProcessor()) // This must be added before InMemoryExporter
+                .AddInMemoryExporter(exportedItems)
+                .Build();
+
+            // The above line is required only in applications
+            // which decide to use OpenTelemetry.
+
+            // Libraries would simply write the following lines of code to
+            // emit activities, which are the .NET representation of OpenTelemetry Spans.
+            var source = new ActivitySource("MyCompany.MyProduct.MyWebServer");
```

Comment:
Most of these logic seems to be duplicated with other exporter examples? Is there a way to de-dupe?

Reply:
https://github.com/open-telemetry/opentelemetry-dotnet/blob/master/examples/Console/InstrumentationWithActivitySource.cs we could move to this. Its already used by otlp and other exporters, so perhaps a good idea to move everything to use this.
(the docs/trace/getting-started can keep the absolute simple example)

Reply:
Should we add InMemoryExporter to the docs/trace/getting-started example?

Reply:
No.